ipamd/ipamd.go

```diff
@@ -243,21 +248,23 @@ func (c *IPAMContext) getLocalPodsWithRetry() ([]*k8sapi.K8SPodInfo, error) {
 		return nil, errors.New("unable to get local pods, giving up")
 	}
 
-	containers, err := c.dockerClient.GetRunningContainers()
+	var containers map[string]*docker.ContainerInfo
 
-	if err != nil {
-		log.Errorf("Not able to get container info from docker: %v", err)
-		return nil, errors.Wrapf(err, "not able to get container info from docker")
+	for retry := 1; retry <= maxK8SRetries; retry++ {
+		containers, err = c.dockerClient.GetRunningContainers()
+		if err == nil {
+			break
+		}
+		log.Infof("Not able to get local containers yet (attempt %d/%d): %v", retry, maxK8SRetries, err)
+		time.Sleep(retryK8SInterval)
 	}
 
```

Comment:
What happens if we exceed maxK8sRetries?  The loop below will iterate over the pods, but there are no containers so it will do nothing.  The function no longer returns an error if it cannot get container info from docker.